users/tests.py

```diff
@@ -0,0 +1,155 @@
+    def test_that_user_should_not_hold_same_email_as_another_user(self):
+        old_user = CustomUser.objects.create(
+            email = "testuser@example.com",
+            password = 'newuserpasswd',
+        )
+        new_user = CustomUser(
+            email = "testuser@example.com",
+            password = 'newuserpasswd',
+        )
+        with self.assertRaisesRegexp(
+            ValidationError,
+            "User with this Email address already exists.",
+        ):
+            new_user.full_clean()
+
+    def test_that_user_should_not_hold_email_without_at_sign(self):
+        new_user = CustomUser(
+            email = "wrongtestuserexample.com",
+            password = 'newuserpasswd',
+        )
+        with self.assertRaisesRegexp(
+            ValidationError,
+            str(EmailValidator.message),
+        ):
+            new_user.full_clean()
+
+    def test_that_user_should_not_hold_first_name_longer_than_FIRST_NAME_MAX_LENGTH(self):
+        new_user = CustomUser(
+            email = "testuser@example.com",
+            password = 'newuserpasswd',
+            first_name = 'a'*(constants.FIRST_NAME_MAX_LENGTH+1),
+        )
+        with self.assertRaisesRegexp(
+            ValidationError,
+            str(MaxLengthValidator.message),
+        ):
+            new_user.full_clean()
+
+    def test_that_user_should_not_hold_last_name_longer_than_LAST_NAME_MAX_LENGTH(self):
+        new_user = CustomUser(
+            email = "testuser@example.com",
+            password = 'newuserpasswd',
+            last_name = 'a'*(constants.LAST_NAME_MAX_LENGTH+1),
+        )
+        with self.assertRaisesRegexp(
+            ValidationError,
+            str(MaxLengthValidator.message),
+        ):
+            new_user.full_clean()
+
+
+class TestCustomUserModelMethods(TestCase):
+    def setUp(self):
+        new_user = CustomUser.objects.create(
+            email="testuser@example.com",
+            first_name="testusername",
+            last_name="testuserlastname",
+        )
+
+    def test_that_get_absolute_url_method_should_return_absolute_url_with_users_email(self):
+        email = "testuser@example.com"
+        search_user = CustomUser.objects.get(email=email)
+        self.assertEqual(
+            search_user.get_absolute_url(),
+            f"/users/{email}/",
+        )
+
+    def test_that_get_full_name_method_should_return_first_and_last_user_name_with_space_between(self):
+        email = "testuser@example.com"
+        search_user = CustomUser.objects.get(email=email)
+        self.assertEqual(
+            search_user.get_full_name(),
+            f"{search_user.first_name} {search_user.last_name}",
+        )
+
+    def test_that_get_short_name_method_should_return_user_first_name(self):
+        email = "testuser@example.com"
+        search_user = CustomUser.objects.get(email=email)
+        self.assertEqual(
+            search_user.get_short_name(),
+            search_user.first_name,
+        )
```

Comment:
Change tests to pytests parametrize please

Reply:
Changing tests to pytests needs @maciejSamerdak test coverage. To use it I would have to rebase to origin master, but during request I should not. Am I wrong?

Reply:
I think pull request isn't nailed to specific commit - we just rate your changes, not changes that are on branches below. Unless rebasing causes bugs - then your code is not working and then you should create some separate commits fixing these.
From my side - you can rebase.